expo-zustand-styled-components/src/components/Profile/Repositories/Repositories.tsx

```diff
@@ -40,20 +39,18 @@ const Repositories = ({ username }: { username: string }) => {
   }, [username]);
 
   useEffect(() => {
-    useRepoSortFilter(userRepos, search)
-  }, [search, userRepos]);
-
-
+    useRepoSortFilter(userRepos, search, filterType, sortBy, language);
+  }, [search, userRepos, filterType]);
 
   return (
     <ContainerStyled
       style={{ justifyContent: isLoading || error ? 'center' : 'flex-start' }}
       screenWidth={width}>
+      <RepoFilter languages={languages} filteredRepoCount={result.length} repoBtnText="New" />
       {isLoading || error ? (
         <LoaderErrorView error={error} />
       ) : (
         <ContentViewStyled>
-          <RepoFilter languages={[]} filteredRepoCount={result.length} repoBtnText="New" />
           <ReposContainer>
```

Comment:
Why is RepoFilter component moved above? It requires the data to be fetched I think?

Reply:
I was trying one thing and I just forgot to move that component back into the right place 😅